test/unit-tests/android-application-manager.ts

```diff
@@ -0,0 +1,129 @@
+			frontOfTask=true task=TaskRecord{fe592ac #449 A=org.nativescript.testApp U=0 StackId=1 sz=1}"
+		];
+
+	public async executeShellCommand(args: string[]): Promise<any> {
+		if (args && args.length > 0) {
+			if (args[0] === "pm") {
+				const passedIdentifier = args[2];
+				if (passedIdentifier === invalidIdentifier) {
+					return Promise.resolve("invalid output string");
+				} else {
+					const testString = this.validTestInput[AndroidDebugBridgeStub.methodCallCount];
+					return Promise.resolve(testString);
+				}
+			} else {
+				this.startedWithActivityManager = this.checkIfStartedWithActivityManager(args);
+				if (this.startedWithActivityManager) {
+					this.validIdentifierPassed = this.checkIfValidIdentifierPassed(args);
+				}
+			}
+		}
+		AndroidDebugBridgeStub.methodCallCount++;
+		Promise.resolve();
+	}
+
+	public getInputLength(): number {
+		return this.validTestInput.length;
+	}
+
+	private checkIfStartedWithActivityManager(args: string[]): Boolean {
+		const firstArgument = args[0].trim();
+		switch (firstArgument) {
+			case "am": return true;
+			case "monkey": return false;
+			default: return false;
+		}
+	}
+
+	private checkIfValidIdentifierPassed(args: string[]): Boolean {
+		if (args && args.length) {
+			const possibleIdentifier = args[args.length - 1];
+			let validTestString = this.expectedValidTestInput[AndroidDebugBridgeStub.methodCallCount];
+			if (possibleIdentifier === validTestString) {
+				return true;
+			}
+		}
+		return false;
+	}
+}
+
+function createTestInjector(): IInjector {
+	let testInjector = new Yok();
+	testInjector.register("androidApplicationManager", AndroidApplicationManager);
+	testInjector.register("adb", AndroidDebugBridgeStub);
+	testInjector.register('childProcess', {});
+	testInjector.register("logger", CommonLoggerStub);
+	testInjector.register("config", {});
+	testInjector.register("staticConfig", {});
+	testInjector.register("androidDebugBridgeResultHandler", {});
+	testInjector.register("options", {justlaunch: true});
+	testInjector.register("errors", {});
+	testInjector.register("identifier", {});
+	testInjector.register("logcatHelper", {});
+	testInjector.register("androidProcessService", {});
+	testInjector.register("httpClient", {});
+	testInjector.register("deviceLogProvider", {});
+	testInjector.register("hooksService", {});
+	return testInjector;
+}
+
+describe("android-application-manager", () => {
+
+	let testInjector: IInjector,
+		androidApplicationManager:AndroidApplicationManager,
+		androidDebugBridge:AndroidDebugBridgeStub;
+
+	beforeEach(() => {
+		testInjector = createTestInjector();
+		androidApplicationManager = testInjector.resolve("androidApplicationManager");
+		androidDebugBridge = testInjector.resolve("adb");
+	});
+	describe("startApplication", () => {
+		it("fires up the right application", async () => {
+			for (let i = 0; i < androidDebugBridge.getInputLength(); i += 1) {
```

Comment:
Well, that's not entirely correct. Consider the following example.
```JavaScript
const iterations = 100*1000*1000;

console.time("i+=1");
for(let i = 0; i < iterations; i+=1) {
}
console.timeEnd("i+=1");

console.time("i++");
for(let i = 0; i < iterations; i++) {
}
console.timeEnd("i++");
```

And the result is:
```
$ node b.js
i+=1: 1118.667ms
i++: 887.539ms

$ node b.js
i+=1: 1116.609ms
i++: 877.461ms

$ node b.js
i+=1: 1106.636ms
i++: 882.537ms
```

But anyway, the codebase contains many lines with `<variable name>++` , so I prefer consistency here.